pkg/controller/windowsmachineconfig/nodeconfig/nodeconfig.go

```diff
@@ -324,6 +342,29 @@ func (nc *nodeConfig) waitForNodeAnnotation(annotation string) error {
 	return nil
 }
 
+// configureCNI populates the CNI config template and sends the config file location
+// for completing CNI configuration in the windows VM
+func (nc *nodeConfig) configureCNI() error {
+	// set the hostSubnet value in the network struct
+	if err := nc.network.setHostSubnet(nc.node.Annotations[HybridOverlaySubnet]); err != nil {
+		return errors.Wrapf(err, "error populating host subnet in node network")
+	}
+	// populate the CNI config file with the host subnet and the service network CIDR
+	configFile, err := nc.network.populateCniConfig(nc.clusterServiceCIDR, wkl.CNIConfigTemplatePath)
```

Comment:
While this approach is not wrong, ideally you'd pass finalizer for deleting the file you created. An example would be to return `cleanUp` function which deletes the configFile that was created.

Reply:
returning a cleanup function would need it to be deferred at the receiving end if I am not wrong. I thought we are trying to avoid using defer here.

Reply:
Not sure why you have to use defer with a cleanup function.